### PR TITLE
NMNIST integrity check and torch labels

### PR DESCRIPTION
**Pull request by Jegp**
This PR addresses #117 
It also converts the label list to torch tensors in `utils.pad_tensors`, which is needed for pytorch loss functions

### CODE REVIEW

**Comment by biphasic** [comment]
very nice! thank you!
